db/src/main/kotlin/edu/artic/ResultExtensions.kt

```diff
@@ -16,19 +16,21 @@ import org.json.JSONObject
  * Looks for error, message and detail keys in order.
  */
 fun Result<*>.getErrorMessage(): String {
+    val SOME_THING_WENT_WRONG : String = "Something went wrong"
+
     lateinit var errorMessage: String
     try {
         this.response().errorBody()?.also { errorBody ->
             val errorJSON = JSONObject(errorBody.string())
-            errorMessage = errorJSON.optString("error")
+            val message = errorJSON.optString("error")
                     .orIfNullOrBlank(errorJSON.optString("message"))
                     .orIfNullOrBlank(errorJSON.optString("Message"))
                     .orIfNullOrBlank(errorJSON.optString("detail"))
-                    ?: "Something went wrong"
```

Comment:
What are your thoughts on using `.orIfNullOrBlank(SOME_THING_WENT_WRONG)` here?

Reply:
`orIfNullOrBlank` returns `String?` and `errorMessage` is of type `String`.

Reply:
I can use bang operator here if i use `.orIfNullOrBlank(SOME_THING_WENT_WRONG)`
```
val message = errorJSON.optString("error")
                    .orIfNullOrBlank(errorJSON.optString("message"))
                    .orIfNullOrBlank(errorJSON.optString("Message"))
                    .orIfNullOrBlank(errorJSON.optString("detail"))
                    .orIfNullOrBlank(SOMETHING_WENT_WRONG)
            errorMessage = message!!
```

Reply:
Ok.